src/lib.rs

```diff
@@ -1,66 +1,248 @@
-#![allow(non_snake_case)]
-
 //!
 //! Parsing of ABNF Core Rules
 //!
-//! See https://tools.ietf.org/html/rfc5234#appendix-B.1
+//! See <https://tools.ietf.org/html/rfc5234#appendix-B.1>
 //!
 
 pub mod complete;
 pub mod streaming;
 
-pub fn is_alpha(c: char) -> bool {
-    c.is_ascii_alphabetic()
+use nom::AsChar;
+
+/// A-Z / a-z
+///
+/// ALPHA = %x41-5A / %x61-7A
+pub fn is_alpha(c: impl AsChar) -> bool {
+    matches!(c.as_char(), '\x41'..='\x5A' | '\x61'..='\x7A')
+}
+
+#[test]
+fn test_is_alpha() {
+    assert!(is_alpha(b'a'));
+    assert!(is_alpha('A'));
+    assert!(is_alpha('z'));
+    assert!(is_alpha('Z'));
+    assert!(!is_alpha('0'));
+    assert!(!is_alpha('9'));
+    assert!(!is_alpha('#'));
+}
+
+/// BIT = "0" / "1"
+/// BIT = %x30 / %x31
+pub fn is_bit(c: impl AsChar) -> bool {
+    matches!(c.as_char(), '\x30'..='\x31')
+}
```

Comment:
I added myself `BIT = %x30 / %x31` cause the ABNF use string representation but doesn't define `"0"` and `"1"` directly.

The unit test use rust char so the value are checked.